docs/developer/eventing/sources/ping-source/README.md

```diff
@@ -1,154 +1,226 @@
+            targetPort: 8080
+        ```
 
-1. Create a PingSource that sends an event containing `{"message": "Hello world!"}` every minute:
+    1. Apply the YAML file by running the command:
 
-    === "YAML"
+        ```bash
+        kubectl apply -f <filename>.yaml
+        ```
+        Where `<filename>` is the name of the file you created in the previous step.
 
-        1. Copy the following YAML into a file:
+1. Create the PingSource object.
 
-            ```yaml
-            apiVersion: sources.knative.dev/v1
-            kind: PingSource
-            metadata:
-              name: test-ping-source
-            spec:
-              schedule: "*/1 * * * *"
-              contentType: "application/json"
-              data: '{"message": "Hello world!"}'
-              sink:
-                ref:
-                  apiVersion: v1
-                  kind: Service
-                  name: event-display
-            ```
+    !!! note
+        The data you want to send must be represented as text in the PingSource YAML file.
+        Events that send binary data cannot be directly serialized in YAML.
+        However, you can send binary data that is base64 encoded by using
+        `dataBase64` in place of `data` in the PingSource spec.
+
+    Use one of the following options:
 
-        1. Apply the YAML file by running the command:
+    === "kn"
+
+        - To create a PingSource that sends data that can be represented as
+        plain text, such as text, JSON, or XML, run the command:
 
             ```bash
-            kubectl apply -f <filename>.yaml
+            kn source ping create <pingsource-name> \
+              --namespace <namespace> \
+              --schedule "<cron-schedule>" \
+              --data '<data>' \
+              --sink <sink-name>
             ```
-            Where `<filename>` is the name of the file you created in the previous step.
+            Where:
 
-    === "kn"
+            - `<pingsource-name>` is the name of the PingSource that you want to create, for example, `test-ping-source`.
+            - `<namespace>` is the name of the namespace that you created in step 1 above.
+            - `<cron-schedule>` is a cron expression for the schedule for the PingSource to send events, for example, `*/1 * * * *` sends an event every minute.
+            - `<data>` is the data you want to send. This data must be represented as text, not binary. For example, a JSON object such as `{"message": "Hello world!"}`.
+            - `<sink-name>` is the name of your sink, for example, `http://event-display.pingsource-example.svc.cluster.local`.
+
+            For a list of available options, see the [Knative client documentation](https://github.com/knative/client/blob/main/docs/cmd/kn_source_ping_create.md).
+
+    === "YAML"
+
+        - To create a PingSource that sends data that can be represented as plain text,
+        such as text, JSON, or XML:
+
+            1. Create a YAML file using the template below:
```

Comment:
does `kn` support binary data?

Reply:
I guessed that is does. I'm not sure if I'm right though. 

I looked at the [kn docs](https://github.com/knative/client/blob/main/docs/cmd/kn_source_ping_create.md#options) for the `kn source ping create` command and I found the option `--encoding base64` which I assumed was for users to apply binary data encoded in base64. 

 See line 167 in this PR for where I added this option.

Reply:
@lionelvillard do you think I should remove the `kn` command for binary data?

Reply:
sounds good.